pgmpy/estimators/ScoreCache.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+from pgmpy.estimators import StructureScore
+
+
+class ScoreCache(StructureScore):
+
+    def __init__(self, base_scorer, data, max_size=10000, **kwargs):
+        """
+        A wrapper class for StructureScore instances, which implement a decomposable score,
+        that caches local scores.
+        Based on the global decomposition property of Bayesian networks for decomposable scores.
+
+        Parameters
+        ----------
+        :param base_scorer: StructureScore instance
```

Comment:
@julianreichinger This still doesn't seem right. I am sorry for not being clearer in my last comment. You will need to remove the `:param` from the front for numpy style docstring. You can also have a look at it in more detail here: http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html.